classes/filter/FilterDAO.inc.php

```diff
@@ -312,7 +312,9 @@ function getObjectsByGroup($groupSymbolic, $contextId = CONTEXT_ID_NONE, $getTem
 		//    result set that comply with the current runtime
 		//    environment.
 		$matchingFilters = array();
-		foreach($result->GetAssoc() as $filterRow) {
+		$isSqlServer = Config::getVar('database', 'ms_sql');
+		$rows = $isSqlServer ? $result->GetRows() : $result->GetAssoc();
```

Comment:
Since we're not using the key in the `foreach` loop, I think we can just use `GetRows` for all drivers.

Reply:
So I can remove the condition and use GetRows for every databases. Can you confirm to me that this change won't cause some problems in the subsequent functions?

Reply:
Yes, in this case we're not using the returned array key at all.